public/app/core/services/backend_srv.ts

```diff
@@ -103,6 +103,34 @@ export class BackendSrv implements BackendService {
     return await this.request({ method: 'PUT', url, data });
   }
 
+  augmentRequestHeaders(headers: BackendSrvRequest['headers']) {
+    let augmentedHeaders: BackendSrvRequest['headers'] = headers;
+    const orgId = this.dependencies.contextSrv.user?.orgId;
+
+    if (orgId) {
+      augmentedHeaders = augmentedHeaders || {};
+      augmentedHeaders['X-Grafana-Org-Id'] = orgId;
+    }
+
+    return augmentedHeaders;
+  }
+
+  augmentDataSourceRequestHeaders(headers: BackendSrvRequest['headers']) {
+    let augmentedHeaders: BackendSrvRequest['headers'] = this.augmentRequestHeaders(headers);
+
+    if (augmentedHeaders?.Authorization) {
+      augmentedHeaders['X-DS-Authorization'] = augmentedHeaders.Authorization;
+      delete augmentedHeaders.Authorization;
+    }
+
+    if (this.noBackendCache) {
+      augmentedHeaders = augmentedHeaders || {};
+      augmentedHeaders['X-Grafana-NoCache'] = 'true';
+    }
+
+    return augmentedHeaders;
+  }
+
```

Comment:
Great work on the refactoring of this, did you see https://github.com/grafana/grafana/blob/02140fff2d15004e7310fa01663610797a3b90b8/public/app/core/utils/fetch.ts#L4 because we do a lot of header specific work there. Not sure you need it though, just mention it.

Reply:
Thanks!
IIRC (since all of this is up in the air right now, I won't look at it in detail), I did indeed see this because it is exactly where my "hunt" for headers started and where e.g. `augmentRequestHeaders()` is eventually used if you follow the chain of calls.